### PR TITLE
feat: improve EAS Update workflow to build dev client when fingerprint changes

### LINKED ISSUE #54: Improve EAS Update to build DevClient when fingerprint changes

### PR DESCRIPTION
**Pull request by km1chno**
Resolves #54

### CODE REVIEW

File: src/recipes/build-release.ts [hunk 1]
```diff
@@ -71,19 +71,17 @@ export const createReleaseBuildWorkflows = async (
...
-    )
+    toolbox.expo.prebuild(context, { cleanAfter: false })
   }
 
   const iOSAppName = toolbox.filesystem
     .list('ios')
     ?.find((file) => file.endsWith('.xcworkspace'))
     ?.replace('.xcworkspace', '')
 
+  if (!existsAndroidDir) toolbox.filesystem.remove('android')
```
Comment (maciekstosio) [inline src/recipes/build-release.ts:82]: I'd add some comment or rename the variable - initially, I thought that you're removing folder if it doesn't exists - which odesn't make sense :D And what we're doing is we're removing the folder if user didn't had it/modified it before

File: src/recipes/eas.ts [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+    },
+  }
+
+  if (!withIOSCredentials) {
+    patch.build.development['ios'] = {
+      simulator: true,
+    }
+  }
+
+  await toolbox.patching.update('eas.json', (config) => {
```
Comment (maciekstosio) [inline src/recipes/eas.ts:32]:
```suggestion
  await toolbox.patching.update('eas.json', (config) => recursiveAssign(config, patch))
```

**Review by maciekstosio — COMMENTED** [review]
Two minor comments, rest looks good

File: src/recipes/eas.ts (hunk 2 above)
Comment (km1chno) [reply src/recipes/eas.ts]: nice, changed in d2aad9439aab993dceaa7880f622a4f7a3f1b78e

File: src/recipes/build-release.ts (hunk 1 above)
Comment (km1chno) [reply src/recipes/build-release.ts]: yes it is a bit misleading:D changed to `clean(Android/IOS)AfterPrebuild` in d2aad9439aab993dceaa7880f622a4f7a3f1b78e

**Review by maciekstosio — DISMISSED** [review]

**Review by maciekstosio — APPROVED** [review]

**Pull request merged**